python/test/unit/ampere/test_gemm_mixed_dtype_small_tile.py

```diff
@@ -0,0 +1,152 @@
+import pytest
+import torch
+from torch.testing import assert_close
+
+import triton
+import triton.language as tl
+
+
+@triton.jit
+def matmul_kernel_16bx8b(lhs_ptr,  # (M, K)
+                         rhs_ptr,  # (K, N)
+                         out_ptr,  # (M, N)
+                         # shape information (strides)
+                         M, N, K,
+                         # block information
+                         block_m: tl.constexpr, block_n: tl.constexpr, block_k: tl.constexpr, dtype_a: tl.constexpr,
+                         dtype_b: tl.constexpr):
+    start_m = tl.program_id(0)  # start (axis m)
+    start_n = tl.program_id(1)  # start (axis n)
+
+    acc = tl.zeros([block_m, block_n], dtype=tl.float32)
+
+    for start_k in range(0, K, block_k):
+        lhs_tile = (start_m * block_m + tl.arange(0, block_m))[:, None] * K + \
+            (start_k + tl.arange(0, block_k))[None, :]
+
+        rhs_tile = (start_k + tl.arange(0, block_k))[:, None] * N + \
+            (start_n * block_n + tl.arange(0, block_n))[None, :]
+
+        lhs_mask = ((start_m * block_m + tl.arange(0, block_m)) < M)[:, None] \
+            * ((start_k + tl.arange(0, block_k)) < K)[None, :]
+
+        rhs_mask = ((start_n * block_n + tl.arange(0, block_n)) < N)[None, :] \
+            * ((start_k + tl.arange(0, block_k)) < K)[:, None]
+
+        lhs = tl.load(lhs_ptr + lhs_tile, mask=lhs_mask, other=0.0)
+        rhs = tl.load(rhs_ptr + rhs_tile, mask=rhs_mask, other=0.0)
+
+        if dtype_a == torch.float16:
+            acc = tl.dot(lhs, rhs.to(tl.float16), acc)
+        elif dtype_a == torch.bfloat16:
+            acc = tl.dot(lhs, rhs.to(tl.bfloat16), acc)
+
+    out_tile = (start_m * block_m + tl.arange(0, block_m))[:, None] * N + \
+        start_n * block_n + tl.arange(0, block_n)[None, :]
+
+    mask = ((start_m * block_m + tl.arange(0, block_m)) < M)[:, None] & \
+        ((start_n * block_n + tl.arange(0, block_n)) < N)[None, :]
+
+    # print("acc ", acc)
+
+    tl.store(out_ptr + out_tile, acc, mask=mask)
+
+
+# this are the test cases for mixed dtype GEMM (16-bit and 8-bit) when the tile size is small (e.g., 16)
+@pytest.mark.parametrize('M,N,K,block_m,block_n,block_k,dtype_a,dtype_b', [
+    (256, 256, 256, block_m, block_n, block_k, dtype_a, dtype_b)
+    for block_m in [16, 32, 64, 128]  # block size m
+    for block_n in [16, 32, 64, 128]  # block size n
+    for block_k in [16, 32]  # block size k
+    for dtype_a in [torch.float16, torch.bfloat16]  # loop over different data types of 16-bit
```

Comment:
We have added the test cases for the different 8-bit data types and different 16-bit data types.